add gh-actions and husky

technophile-04 opened the pull request:
# Description

Added GitHub workflow and husky git hook `pre-commit` scoped to **`packages/react-app`** only. 
### Local flow  : 
Contributor fork/clones the repo makes some `changes`, `stage` them and then `commits`. Now before committing,  following  scripts run : 
- **`prettier --write`** -> Runs prettier on all the `staged` files.
- **`eslint --fix`** -> Fixes the `eslint` issue which are fixable . 
- **`eslint --max-warnings=0 --ignore-path packages/react-app/.eslintignore`** -> Runs `eslint` checks and don't allow  to `commit` if there are any `errors` or `warnings`.  (`--max-warnings=0`  throws error if there are any `warnings` aslo this way if there are any warnings you won't be able to commit ) . 

pre-commit husky hook running all above scripts : 
![Screenshot 2022-08-31 at 7 45 48 PM](https://user-images.githubusercontent.com/80153681/187700904-b7be6582-f2c3-49fc-807b-b7e5332aa788.jpg)

Once checks are done : 
![Screenshot 2022-08-31 at 7 46 02 PM](https://user-images.githubusercontent.com/80153681/187700916-e5fe66ac-3e14-49a5-b9cf-189a789289c1.jpg)

### Github workflow : 
When `changes` are pushed to `master` branch or `pull request` is made to `master` branch following scirpt is run : 
- **`eslint ./src/**/*.{js,jsx} --ignore-path ./.eslintignore --max-warnings=0 && eslint ./src/*.{js,jsx} --ignore-path ./.eslintignore --max-warnings=0`** ->  Runs `eslint` checks and `fails` if there are any `errors` or `warnings`.  (`--max-warnings=0`  throws error if there are any `warnings`) . 

PS : The above script can be improved I have used `&& eslint ./src/*.{js,jsx}` to include `App.jsx`, `index.jsx` (files which are direct child of `src`). Don't know why `eslint ./src/**/*.{js,jsx}` wast not including `App.jsx` and files parallel to it. 

Github actions running : 
![Screenshot 2022-08-31 at 7 43 55 PM](https://user-images.githubusercontent.com/80153681/187701402-97f30910-450c-441d-8800-4fb34a76758e.jpg)



Fixes #861 

## Note : 
I have added `lint-staged` dependency to `root` for local husky `pre-commit` flow. 

**PS : There many file changes because I fixed some of the `eslint` warnings now there are no warnings `packages/react-app`. I have added missing dependency variables in `useEffect` to `packages/react-app/src/hooks/TokenList.js` need to `test` it. Rest all changes are `prettier` changes.** 

Open for suggestions and bad code quality/practices criticism  😅 . Will try my best to improve 🙌

carletex commented on packages/react-app/src/hooks/TokenList.js:
```diff
@@ -41,7 +41,7 @@ const useTokenList = (tokenListUri, chainId) => {
     };
     getTokenList();
-  }, [tokenListUri]);
+  }, [tokenListUri, _tokenListUri, chainId]);
```
It seems this hook is not being used:

`views/Hints.jsx` uses the `useTokenList` hook but from `eth-hooks/dapps/dex`, not this one.

I tested it anyway, and it works ;)

carletex reviewed (commented):
Hey @technophile-04 ! This is great stuff ;)

I reviewed & tested it. Check out the review comments.

Good work!!!

technophile-04 commented on package.json:
```diff
@@ -68,6 +70,14 @@
+    "packages/react-app/src/**/*.{js,jsx}": [
+      "prettier --write",
+      "eslint --fix",
+      "eslint --max-warnings=0 --ignore-path packages/react-app/.eslintignore"
```
Yes completely agree with you, We don't want to disturb the local flow...Regarding the `two options` , After some digging, trial, and testing, you are correct `line 80`  is completely **redundant**  after removing `--max-warnings=0` so I am removing `line 80`. Also `lint-staged` only shows `warnings` if there is an `error` in `linting` : [How to show warnings? #671](https://github.com/okonet/lint-staged/issues/671) . So will be adding `--verbose` in the husky `pre-commit` hook to show the warnings .

technophile-04 commented:
Added new commit [faf7b1f](https://github.com/scaffold-eth/scaffold-eth/pull/864/commits/faf7b1f1612fe7e2d311ac279a526416318c52bb) 
- Removed the both unnecessary scripts and lines . 
- Added `--verbose` flag for just showing the warnings while `committing` in  husky `pre-commit`  : 

with `--verbose` flag : 
![Screenshot 2022-09-01 at 4 58 55 PM](https://user-images.githubusercontent.com/80153681/187907057-58cd7e1b-f385-4ac1-82d0-f28569cac037.jpg)

without flag (It don't show any warnings even if there are warnings) :
![Screenshot 2022-09-01 at 4 59 36 PM](https://user-images.githubusercontent.com/80153681/187907188-a9626e71-46d9-4ce1-b69e-cc891ea6b6b1.jpg)

carletex commented:
New lint tests passing:
![image](https://user-images.githubusercontent.com/2486142/187943181-a091491c-6663-4d24-a3fc-788aff858551.png)
